lib/rack/test.rb

```diff
@@ -234,6 +234,23 @@ def follow_redirect!
         )
       end
 
+      # Yield to the block, and restore the last request, last response, and
+      # cookie jar to the state they were prior to block execution upon
+      # exiting the block.
+      def restore_state
```

Comment:
:thinking: I'm the worst person to provide hints for better naming :sob:, but my initial idea was to call this method `preserve_state` or `keep_state`.

Reply:
OpenGL used to have push state and pop state functions. That was fun...

```
  glPushMatrix();
  glRotated((GLdouble) Vangle,1.0,0.0,0.0);
  gluLookAt (0.0, 0.0, 5.0,   0.0, 0.0, 0.0,    0.0, 1.0, 0.0);
  glPopMatrix();
```

I think something along the lines of `isolated_request` or `push_request_state` would communicate better the operation being done.

It might be nice to think about whether operations like `get` can have this too:

e.g.

```
get "/" do
  # last_response only available here.
end